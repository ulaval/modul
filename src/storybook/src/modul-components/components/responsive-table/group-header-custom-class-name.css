.test-group-header-custom-class-name.test-group-header-custom-class-name {
    border-left: 4px solid #000;
    background: #ccc;
}

.test-group-header-custom-class-name.test-group-header-custom-class-name th {
    padding: 8px 16px;
}

.test-icon-accordion-custom-class-name.test-icon-accordion-custom-class-name {
    color: blue;
}

.test-cell-custom-class-name.test-cell-custom-class-name {
    background: #000;
    color: #fff;
}
