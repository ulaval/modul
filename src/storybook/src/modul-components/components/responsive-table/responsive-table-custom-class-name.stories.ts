import { select } from '@storybook/addon-knobs';
import { RESPONSIVE_TABLE_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableGroupHeaderStyle, MTableRowsStyle } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import './group-header-custom-class-name.css';
import { DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE, DEFAULT_TABLE_ROW_1, DEFAULT_TABLE_ROW_2, DEFAULT_TABLE_ROW_3 } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/Custom class-name`,
    parameters: { fileName: __filename }
};

export const GroupHeader = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: [
            {
                name: 'CustomStyle',
                header: {
                    title: 'Group #1',
                    className: 'test-group-header-custom-class-name'
                },
                accordion: {
                    open: true,
                    iconClassName: 'test-icon-accordion-custom-class-name'
                },
                rows: [
                    DEFAULT_TABLE_ROW_1,
                    DEFAULT_TABLE_ROW_2,
                    DEFAULT_TABLE_ROW_3
                ]
            }
        ]
    }),
    props: {
        groupHeaderStyle: {
            default: select(
                'Prop group-header-style',
                Enums.toValueArray(MTableGroupHeaderStyle),
                MTableGroupHeaderStyle.Any
            )
        },
        rowsStyle: {
            default: select(
                'Prop rows-style',
                Enums.toValueArray(MTableRowsStyle),
                MTableRowsStyle.Borders
            )
        }
    },
    template: `<${RESPONSIVE_TABLE_NAME}
        id="GroupHeader"
        :columns="columns"
        :row-groups="rowGroups"
        :group-header-style="groupHeaderStyle"
        :rows-style="rowsStyle"
    />`
});

export const Cells = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rows: [
            {
                cells: {
                    name: {
                        value: 'Vincent Guilmette',
                        className: 'add-custom-class-cell',
                        isHeader: true
                    },
                    id: {
                        value: '111 222 333',
                        className: 'test-cell-custom-class-name'
                    },
                    email: {
                        value: 'vincent.gui@gmail.com'
                    },
                    dateOfBirth: {
                        value: '1976-12-05'
                    },
                    phoneNumber: {
                        value: '819 232-3754'
                    }
                }
            }
        ]
    }),
    props: {
        rowsStyle: {
            default: select(
                'Prop rows-style',
                Enums.toValueArray(MTableRowsStyle),
                MTableRowsStyle.Borders
            )
        }
    },
    template: `<${RESPONSIVE_TABLE_NAME}
        id="Cells"
        :columns="columns"
        :rows="rows"
        :rows-style="rowsStyle"
    />`
});
