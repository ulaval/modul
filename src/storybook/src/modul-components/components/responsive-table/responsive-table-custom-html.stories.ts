import { RESPONSIVE_TABLE_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableRowsStyle } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import './group-header-custom-class-name.css';
import { DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE, DEFAULT_TABLE_ROW_GROUPS_2, getScopeSlotTemplate } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/Custom HTML - Scope slot`,
    parameters: { fileName: __filename }
};

export const RowCell = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="RowCell"
        :columns="columns"
        :row-groups="rowGroups"
    >
        <template #row-cell="{ rowsGroup, row, cell }">
            ${ getScopeSlotTemplate('{{ cell.value }} (Slot row-cell)')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

RowCell.story = {
    name: 'row-cell'
};

export const RowCellWithColumn = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="RowCellWithColumn"
        :columns="columns"
        :row-groups="rowGroups"
    >
        <template #row-cell.id="{ rowsGroup, row, cell }">
            ${ getScopeSlotTemplate('{{ cell.value }} (Slot row-cell.id)')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

RowCellWithColumn.story = {
    name: 'row-cell.<column.name>'
};


export const RowCellWithGroupName = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2,
        rowsGroupName: DEFAULT_TABLE_ROW_GROUPS_2[0].name,
        rowsStyle: MTableRowsStyle.Borders
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="RowCellWithGroupName"
        :columns="columns"
        :row-groups="rowGroups"
        :rows-style="rowsStyle"
    >
        <template
            :slot="'row-cell.id.' + rowsGroupName"
            slot-scope="{ rowsGroup, row, cell, column }">
            ${ getScopeSlotTemplate('{{ cell.value }} (Slot row-cell.id.{{ rowsGroupName }})')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

RowCellWithGroupName.story = {
    name: 'row-cell.<column.name>.<rowsGroup.name>'
};

export const GroupeHeaderTitle = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2,
        rowsGroupName: DEFAULT_TABLE_ROW_GROUPS_2[1].name,
        rowsStyle: MTableRowsStyle.Borders
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="GroupeHeaderTitle"
        :columns="columns"
        :row-groups="rowGroups"
        :rows-style="rowsStyle"
    >
        <template
            #group-header-title="{ rowsGroup, title, column }">
            ${ getScopeSlotTemplate('{{ title }} (Slot group-header-cell)')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

GroupeHeaderTitle.story = {
    name: 'group-header-title'
};

export const GroupeHeaderTitleWithGroupName = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2,
        rowsGroupName: DEFAULT_TABLE_ROW_GROUPS_2[0].name,
        rowsStyle: MTableRowsStyle.Borders
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="GroupeHeaderTitleWithGroupName"
        :columns="columns"
        :row-groups="rowGroups"
        :rows-style="rowsStyle"
    >
        <template
            :slot="'group-header-title.' + rowsGroupName"
            slot-scope="{ rowsGroup, title, column }">
            ${ getScopeSlotTemplate('{{ title }} (Slot group-header-cell.{{ rowsGroupName }})')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

GroupeHeaderTitleWithGroupName.story = {
    name: 'group-header-title.<rows-group.name>'
};

export const GroupeHeaderCell = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2,
        rowsStyle: MTableRowsStyle.Borders
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="GroupeHeaderCell"
        :columns="columns"
        :row-groups="rowGroups"
        :rows-style="rowsStyle"
    >
        <template #group-header-cell="{ rowsGroup, row, cell, column }">
            ${ getScopeSlotTemplate('{{ cell.value }} (Slot group-header-cell)')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

GroupeHeaderCell.story = {
    name: 'group-header-cell'
};

export const GroupeHeaderCellWithColumn = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2,
        rowsStyle: MTableRowsStyle.Borders
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="GroupeHeaderCellWithColumn"
        :columns="columns"
        :row-groups="rowGroups"
        :rows-style="rowsStyle"
    >
        <template #group-header-cell.id="{ rowsGroup, row, cell, column }">
            ${ getScopeSlotTemplate('{{ cell.value }} (Slot group-header-cell.id)')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

GroupeHeaderCellWithColumn.story = {
    name: 'group-header-cell.<column.name>'
};

export const GroupeHeaderCellWithGroupName = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2,
        rowsGroupName: DEFAULT_TABLE_ROW_GROUPS_2[1].name,
        rowsStyle: MTableRowsStyle.Borders
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="GroupeHeaderCellWithGroupName"
        :columns="columns"
        :row-groups="rowGroups"
        :rows-style="rowsStyle"
    >
        <template
            :slot="'group-header-cell.id.' + rowsGroupName"
            slot-scope="{ rowsGroup, row, cell, column }">
            ${ getScopeSlotTemplate('{{ cell.value }} (Slot group-header-cell.id.{{ rowsGroupName }})')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

GroupeHeaderCellWithGroupName.story = {
    name: 'group-header-cell.<column.name>.<rows-group.name>'
};

export const headCell = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2,
        rowsGroupName: DEFAULT_TABLE_ROW_GROUPS_2[0].name,
        rowsStyle: MTableRowsStyle.Borders
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="headCell"
        :columns="columns"
        :row-groups="rowGroups"
        :rows-style="rowsStyle"
    >
        <template #head-cell="{ column }">
            ${ getScopeSlotTemplate('{{ column.value }} (Slot head-cell)')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

headCell.story = {
    name: 'head-cell'
};

export const headCellWithColumn = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: DEFAULT_TABLE_ROW_GROUPS_2,
        rowsGroupName: DEFAULT_TABLE_ROW_GROUPS_2[0].name,
        rowsStyle: MTableRowsStyle.Borders
    }),
    template: `<${RESPONSIVE_TABLE_NAME}
        id="headCellWithColumn"
        :columns="columns"
        :row-groups="rowGroups"
        :rows-style="rowsStyle"
    >
        <template #head-cell.id="{ column }">
            ${ getScopeSlotTemplate('{{ column.value }} (Slot head-cell.id)')}
        </template>
    </${RESPONSIVE_TABLE_NAME}>`
});

headCellWithColumn.story = {
    name: 'head-cell.<column.name>'
};

