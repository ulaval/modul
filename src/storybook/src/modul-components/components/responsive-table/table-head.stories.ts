import { actions } from '@storybook/addon-actions';
import { boolean, select } from '@storybook/addon-knobs';
import { RESPONSIVE_TABLE_NAME, TABLE_HEAD_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableColumn, MTableHeadRows, MTableHeadStyle } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import Vue, { VueConstructor } from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { COMPLEX_TABLE_HEAD_ROWS, DEFAULT_TABLE_HEAD_ROWS, MAIN_ROW } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/${TABLE_HEAD_NAME}`,
    parameters: { fileName: __filename }
};

const getTemplate: (
    headRows: MTableHeadRows,
    columns: MTableColumn[],
    slotHeadCell?: boolean,
    disabled?: boolean
) => VueConstructor = (
    headRows: MTableHeadRows,
    columns: MTableColumn[],
    slotHeadCell: boolean = false,
    disabled: boolean = false
): VueConstructor => {
    return Vue.extend({
        props: {
            firstColumnFixed: {
                default: boolean('Prop first-column-fixed', false)
            },
            disabled: {
                default: boolean('Prop disabled', disabled)
            },
            headStyle: {
                default: select(
                    'Prop head-style',
                    Enums.toValueArray(MTableHeadStyle),
                    MTableHeadStyle.Light
                )
            },
            slotHeadCell: {
                default: boolean('Slot head-cell', false)
            },
            slotHeadCellWithColumn: {
                default: boolean('Slot head-cell.<column.name>', slotHeadCell)
            }
        },
        data: () => ({
            headRows: headRows,
            columns: columns
        }),
        methods: actions(
            'emitSort'
        ),
        template: `<${TABLE_HEAD_NAME}
        id="testTableHead"
        :head-rows.sync="headRows"
        :head-style="headStyle"
        :first-column-fixed="firstColumnFixed"
        :disabled="disabled"
        @sort="emitSort"
    >
        <template
            v-if="slotHeadCellWithColumn"
            v-for="(column, columnIndex) in columns"
            :slot="'head-cell.' + column.name"
            slot-scope="{ column }"
        >
            <template v-if="column && column.value">
                <m-checkbox
                    v-if="columnIndex == 0"
                    :key="columnIndex"
                >{{column.value}}
                </m-checkbox>
                <template v-else>
                    {{column.value}}
                </template>
                (Slot head-cell.{{column.name}})
            </template>
        </template>
        <template
            v-if="slotHeadCell"
            #head-cell="{ column }">
            {{ column.value }}
            (Slot head-cell)
        </template>
    </${TABLE_HEAD_NAME}>`
    });
};

export const Sandbox = () => {
    return getTemplate(DEFAULT_TABLE_HEAD_ROWS, DEFAULT_TABLE_HEAD_ROWS[MAIN_ROW].columns);
};

export const ComplexeHeadRows = () => {
    return getTemplate(COMPLEX_TABLE_HEAD_ROWS, COMPLEX_TABLE_HEAD_ROWS[MAIN_ROW].columns);
};

export const CustomHTML = () => {
    return getTemplate(DEFAULT_TABLE_HEAD_ROWS, DEFAULT_TABLE_HEAD_ROWS[MAIN_ROW].columns, true);
};

CustomHTML.story = {
    name: 'Custom HTML - Use scope slot'
};

export const PropDisabled = () => {
    return getTemplate(DEFAULT_TABLE_HEAD_ROWS, DEFAULT_TABLE_HEAD_ROWS[MAIN_ROW].columns, false, true);
};



