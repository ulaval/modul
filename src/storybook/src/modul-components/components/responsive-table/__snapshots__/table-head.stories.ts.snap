// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-responsive-table/m-table-head Complexe Head Rows 1`] = `
<thead
  class="m-table-head m--is-head-style-light m--has-multiple-head-row"
>
  <tr>
    <th
      class="m--is-text-align-left"
      id="forlderSpacingColumn-testTableHead"
      rowspan="2"
      scope="col"
      style="background: rgb(255, 255, 255);"
    >
      <!---->
    </th>
    <th
      class="m--is-text-align-center"
      colspan="4"
      id="forlderAColumn-testTableHead"
      scope="col"
    >
      
                        Folder A
                    
    </th>
    <th
      class="m--is-text-align-center"
      colspan="4"
      id="forlderBColumn-testTableHead"
      scope="col"
    >
      
                        Folder B
                    
    </th>
  </tr>
  <tr>
    <th
      class="m--is-text-align-center"
      colspan="2"
      id="beforeForlderAColumn-testTableHead"
      scope="col"
    >
      
                        Before
                    
    </th>
    <th
      class="m--is-text-align-center"
      colspan="2"
      id="afterForlderAColumn-testTableHead"
      scope="col"
    >
      
                        After
                    
    </th>
    <th
      class="m--is-text-align-center"
      colspan="2"
      id="beforeForlderBColumn-testTableHead"
      scope="col"
    >
      
                        Before
                    
    </th>
    <th
      class="m--is-text-align-center"
      colspan="2"
      id="afterForlderBColumn-testTableHead"
      scope="col"
    >
      
                        After
                    
    </th>
  </tr>
  <tr>
    <th
      class="m--is-text-align-center"
      id="userNaimeBeforeFolderA-testTableHead"
      scope="col"
    >
      
                        Id
                    
    </th>
    <th
      class="m--is-text-align-left"
      id="userNameBeforeFolderA-testTableHead"
      scope="col"
    >
      
                        User Name
                    
    </th>
    <th
      class="m--is-text-align-left"
      id="userBeforeFolderA-testTableHead"
      scope="col"
    >
      
                        Name
                    
    </th>
    <th
      class="m--is-text-align-left"
      id="userNameAfterFolderA-testTableHead"
      scope="col"
    >
      
                        User Name
                    
    </th>
    <th
      class="m--is-text-align-left"
      id="userAfterFolderA-testTableHead"
      scope="col"
    >
      
                        Name
                    
    </th>
    <th
      class="m--is-text-align-left"
      id="userNameBeforeFolderB-testTableHead"
      scope="col"
    >
      
                        User Name
                    
    </th>
    <th
      class="m--is-text-align-left"
      id="userBeforeFolderB-testTableHead"
      scope="col"
    >
      
                        Name
                    
    </th>
    <th
      class="m--is-text-align-left"
      id="userNameAfterFolderB-testTableHead"
      scope="col"
    >
      
                        User Name
                    
    </th>
    <th
      class="m--is-text-align-left"
      id="userAfterFolderB-testTableHead"
      scope="col"
    >
      
                        Name
                    
    </th>
  </tr>
</thead>
`;

exports[`Storyshots modul-components|m-responsive-table/m-table-head Custom HTML - Use scope slot 1`] = `
<thead
  class="m-table-head m--is-head-style-light"
>
  <tr>
    <th
      class="m--is-sortable m--is-sorted m--is-text-align-left"
      id="name-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          <div
            class="m-checkbox m--is-checkbox-left"
          >
            <input
              class="m-checkbox__hidden"
              id="mCheckbox-fakeId"
              type="checkbox"
            />
             
            <!---->
             
            <label
              class="m-checkbox__wrap"
              for="mCheckbox-fakeId"
            >
              <span
                aria-hidden="true"
                class="m-checkbox__input"
              >
                <span
                  class="m-checkbox__input__icon"
                />
              </span>
               
              <div
                class="m-checkbox__label"
              >
                Name
                
              </div>
            </label>
             
            <div
              class="m-validation-message m-checkbox__message"
            >
              <!---->
               
              <!---->
            </div>
          </div>
          
                (Slot head-cell.name)
             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-sort-asc"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-text-align-center"
      id="id-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                    Id
                
                (Slot head-cell.id)
             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-text-align-left"
      id="email-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                    Email
                
                (Slot head-cell.email)
             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-text-align-left"
      id="dateOfBirth-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                    Date of birth
                
                (Slot head-cell.dateOfBirth)
             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-text-align-left"
      id="phoneNumber-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                    Phone number
                
                (Slot head-cell.phoneNumber)
             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
  </tr>
</thead>
`;

exports[`Storyshots modul-components|m-responsive-table/m-table-head Prop Disabled 1`] = `
<thead
  class="m-table-head m--is-head-style-light"
>
  <tr>
    <th
      class="m--is-sortable m--is-sorted m--is-disabled m--is-text-align-left"
      id="name-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Name
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-disabled m--is-sort-asc"
            disabled="disabled"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-disabled m--is-text-align-center"
      id="id-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Id
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-disabled"
            disabled="disabled"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-disabled m--is-text-align-left"
      id="email-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Email
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-disabled"
            disabled="disabled"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-disabled m--is-text-align-left"
      id="dateOfBirth-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Date of birth
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-disabled"
            disabled="disabled"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-disabled m--is-text-align-left"
      id="phoneNumber-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Phone number
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-disabled"
            disabled="disabled"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
  </tr>
</thead>
`;

exports[`Storyshots modul-components|m-responsive-table/m-table-head Sandbox 1`] = `
<thead
  class="m-table-head m--is-head-style-light"
>
  <tr>
    <th
      class="m--is-sortable m--is-sorted m--is-text-align-left"
      id="name-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Name
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-sort-asc"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-text-align-center"
      id="id-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Id
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-text-align-left"
      id="email-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Email
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-text-align-left"
      id="dateOfBirth-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Date of birth
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
    <th
      class="m--is-sortable m--is-text-align-left"
      id="phoneNumber-testTableHead"
      scope="col"
    >
      <div
        class="m-table-head__cell"
      >
        <div
          class="m-table-head__cell-content"
        >
          
                                Phone number
                             
          <button
            class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
            style="width: 14px; height: 14px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
      </div>
    </th>
  </tr>
</thead>
`;
